src/resolver/nodeModuleLookup.ts

```diff
@@ -72,14 +72,22 @@ export function parseExistingModulePaths(fileAbsPath: string): string[] {
   return existing;
 }
 
+function isDirectory(path: string) {
+  return fileExists(path) && lstatSync(path).isDirectory();
+}
+
 const CACHED_LOCAL_MODULES: { [key: string]: string | null } = {};
-export function findTargetFolder(props: IResolverProps, name: string): string {
+export function findTargetFolder(props: IResolverProps, name: string): { error: string } | { folder: string, isUser: boolean } {
```

Comment:
This function is where the detection happens, so I modified the return type to include `isUser: boolean`.

Also I made it so that when it returns an error, it returns the error message.

Note: I also modified the return type so that it returns *either* an error or a real result, so that TypeScript warns if you try to access the `folder` prop if you haven't first checked the `error` prop.